posts/wordpress-to-jamstack.md

```diff
@@ -0,0 +1,152 @@
+# Technologies Used
+
+- [NextJS](https://nextjs.org/)
+- [TinaCMS](https://tina.io/)
+- [Vercel](https://vercel.com/)
+- [DataStax Astra](https://www.datastax.com/)
+- [SendGrid](https://sendgrid.com/)
+- [EasyCron](https://www.easycron.com/)
+
+# Where To Start?
+
+I started going through the existing site and marking down all the pages and what content lived on each page. It became apparent pretty early on that there was a lot of duplicate content and that I could trim down the number of pages. One I determined what pages I would use I ended up creating my NextJS app with TinaCMS using their [guide](https://tina.io/guides/nextjs/github/initial-setup/) to get a basic framework setup. TinaCMS allows for real-time editing of your site and gives you the ability to use Markdown or JSON to drive the static content on your pages. The static content for your site is then stored in Github and whenever you make a change and publish it is just a commit to your repo. TinaCMS allows for users to create branches from within the browser so whenever they make changes it can go to a branch and get reviewed before being published. I went with NexJS and TinaCMS instead of a traditional CMS is because I wanted full control of the code and to be able to use more modern frameworks.
+
+In NextJS it gives you the ability to fetch data for static generation using `getStaticProps`. This is where you define the static content that you want your page to use and what the TinaCMS form will use.
+
+```javascript
+export const getStaticProps = async ({ preview, previewData }) => {
+  if (preview) {
+    return getGithubPreviewProps({
+      ...previewData,
+      fileRelativePath: "content/im-new.json",
+      parse: parseJson,
+    });
+  }
+  return {
+    props: {
+      sourceProvider: null,
+      error: null,
+      preview: false,
+      file: {
+        fileRelativePath: "content/im-new.json",
+        data: (await import("../content/im-new.json")).default,
+      },
+    },
+  };
+};
+```
+
+This then allows you to use `file.data` in your component along with the TinaCMS form React hook to link your static content with the TinaCMS editor which looks like the image below.
+
+![TinaCMS on website](https://raw.githubusercontent.com/blandfried/blog-usa/master/images/2021/04/tinacms.png)
+
+# Automating Sermon Videos
+
+On the Wordpress site they had to manually update the sermon video section every Monday and I decided that this could be an automated task so they would not have to worry about it. I ended up accomplishing this automation by using the serverless function capabilities that come with using NextJS and hosting it using Vercel. I wrote a Node JS function that calls the YouTube API to retrieve the five latest uploads. To save on api call quotas I made the decision to store the response data in a database since new sermons are uploaded on Sundays and do not change for a week. I figured there was no need to call the YouTube api anytime someone visited the site.
```

Comment:
```suggestion
On the Wordpress site they had to manually update the sermon video section every Monday and I decided that this could be an automated task so they would not have to worry about it. I ended up accomplishing this automation by using the serverless function capabilities that come with using NextJS and hosting it using Vercel. I wrote a Node JS function that calls the YouTube API to retrieve the five latest uploads. To save on API call quotas I made the decision to store the response data in a database since new sermons are uploaded on Sundays and do not change for a week. I figured there was no need to call the YouTube API anytime someone visited the site.
```

Reply:
On the Wordpress, site they had to manually update the sermon video section every Monday.  I determined that this should be an automated task; one less thing for them to worry about. I ended up accomplishing this automation by using the serverless function capabilities that come with using Next.js and hosting it using Vercel. I wrote a Node.js function that calls the YouTube API to retrieve the five latest uploads. To save on API call quotas, I stored the response data in a database since new sermons are uploaded on Sundays and do not change for a week. I figured there was no need to call the YouTube API anytime someone visited the site.